packages/v2/contracts/ANTv2.sol

```diff
@@ -165,8 +176,8 @@ contract ANTv2 is IERC20 {
         _validateSignedData(from, encodedData, v, r, s);
 
         authorizationState[from][nonce] = true;
-        emit AuthorizationUsed(from, nonce);
 
         _transfer(from, to, value);
+        emit AuthorizationUsed(from, nonce);
```

Comment:
@eduadiez Curious on your motivation behind this change. Was this gas-motivated, to avoid spending more gas if the transfer fails, or ordering-related because it made more sense to emit it after the `Transfer` event?

In terms of ordering, I went with ERC-3009's included implementation, but I can kind of see the argument that the authorization is only used up when the transfer is completed.